cartographer/pose_graph/constraint/relative_pose_constraint_2d.cc

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/pose_graph/constraint/relative_pose_constraint_2d.h"
+
+#include "cartographer/common/make_unique.h"
+
+namespace cartographer {
+namespace pose_graph {
+namespace {
+
+template <typename MapType, typename KeyType = typename MapType::key_type,
+          typename ValueType = typename MapType::mapped_type>
+ValueType* FindOrNull(MapType& map, const KeyType& key) {
```

Comment:
Not in this CL but can we have that somewhere in common?

Reply:
Definitely.